base/windows/shop/EquipCompare.ts

```diff
@@ -248,13 +249,14 @@ export class EquipCompare {
                char_current_item = this.data.info.items_list[eq_slots[slot_types[i]].key_name].key_name;
         }
 
-        let atk_diff = this.compare_items(char_current_item, this.selected_item, effect_types.ATTACK, this.selected_char.atk);
-        let def_diff = this.compare_items(char_current_item, this.selected_item, effect_types.DEFENSE, this.selected_char.def);
-        let agi_diff = this.compare_items(char_current_item, this.selected_item, effect_types.AGILITY, this.selected_char.agi);
+        let diffs = {atk: 0, def: 0, agi: 0};
+        let labels = ["ATTACK", "DEFENSE", "AGILITY"];
```

Comment:
Use this: `const labels = [effect_types.ATTACK, effect_types.DEFENSE, effect_types.AGILITY]`. Then, instead of using `effect_types[labels[i]]`, use `labels[i]`. Instead of `diffs[main_stats[labels[i]]]`, use `diff[effect_type_stat[labels[i]]]`.